undertow/WFLY-12473_UUID-instance-id.adoc

```diff
@@ -0,0 +1,65 @@
+= WFLY-12473 UUID instace-id
+:author:            Flavia Rainone
+:email:             frainone@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          instance-id,uuid,uuid-instance-id
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This feature is to be able to set the value of UndertowService.instanceId to a UUID,
+if instance-id was not defined and user desires to have it so.
+EAP6 Web Server would set it as a default whenever instance-id configuration was absent, and Undertow defaults to ${jboss.node.name} instead.
+Some customers would like to have it default to a UUID instead.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12473[WFLY-12473]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1187
+* https://issues.jboss.org/browse/PRODMGT-1934
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+WildFly, Undertow component
+
+=== Other Interested Projects
+
+== Requirements
+
+This will be implemented as an additional attribute under the `subsystem=undertow` management model address.
+
+This will be a simple boolean attribute called `uuid-instance-id` that marks instance-id to be set equal to a uuid.
```

Comment:
@fl4via `instance-id` is set to UUID only if it was not specified? does this implies that `uuid-instance-id` works just when `instance-id` is set to its default value `${jboss.node.name}`?

the PR `|uuid-instance-id | set this to true to indicate that a UUID should be used as the instance id. Can be used only when instance-id is undefined` suggest so, I just wanted a confirm because it's not explicitly stated here

Reply:
@tommaso-borgato, both attributes cannot be defined at the same time, so you're right.

Reply:
@fl4via it took me a while to extract the rule:

```if [[ instance-id == undefined AND uuid-instance-id == true ]] --> UndertowService.instance-id := UUID (--> default route := UUID) ```

* can you confirm that the rule is actually correct?
* given the final customer's goal is to set the default route to UUID (correct me if I am wrong), are we sure we want use such a name as `uuid-instance-id`? are we doing this in order to discourage utilization in customers other than one who asked for this?
* what don't we use something like `instance-id=UUID_UNIVERSALLY_UNIQUE_IDENTIFIER` (or some other string that is unlikely to be used otherwise) to trigger this behavior?